packages/integration_test/lib/integration_test.dart

```diff
@@ -130,35 +132,62 @@ class IntegrationTestWidgetsFlutterBinding
   /// If it's `null`, no extra data is attached to the result.
   ///
   /// The default value is `null`.
-  Map<String, dynamic> reportData;
+  @override
+  Map<String, dynamic> get reportData => _reportData;
+  Map<String, dynamic> _reportData;
+  set reportData(Map<String, dynamic> data) => this._reportData = data;
+
+  /// Manages commands send to driver side.
+  ///
+  /// Only works on Web when tests are run via `flutter driver` command.
+  /// See [WebDriverCommandManager].
+  final DriverCommandManager driverCommandManager =
+      driver_actions.driverCommandManager;
+
+  /// Taking a screenshot.
+  ///
+  /// Called by test methods. Implementation differs for each platform.
+  Future<void> takeScreenshot(String screenshot_name) async {
+    if (kIsWeb) {
```

Comment:
A cleaner approach would be to implement `DriverCommandManager' for the io case. Then you wouldn't need the `if (kIsWeb)`. You'd just always call `driverCommandManager.takeScreenshot` and `driverCommandManager. callbackWithDriverCommands`, and let the platform-specific implementation do the right thing.

Reply:
As discussed offline, we can't add a `DriverCommandManager` for io. But we can change abstraction from `DriverCommandManager` to `CallbackManager`. Then there would be an io implementation as well.